www/activities/diary/js/group.js

```diff
@@ -116,23 +116,42 @@ app.Group = {
     icon.innerText = "add";
     a.appendChild(icon);
 
-    //Energy 
+    //Energy
     const energyUnit = app.Settings.get("units", "energy");
     const energyName = app.Utils.getEnergyUnitName(energyUnit);
 
     let right = document.createElement("div");
     right.className = "margin-horizontal group-energy link icon-only";
-    let value = nutrition[energyName] || 0;
+    let categoryEnergyTotal = nutrition[energyName] || 0;
     let energyUnitSymbol = app.strings["unit-symbols"][energyUnit] || energyUnit;
 
     right.addEventListener("click", function(e) {
       app.Diary.showCategoryNutriments(id, nutrition);
     });
 
-    right.innerText = app.Utils.tidyNumber(Math.round(value), energyUnitSymbol);
+    let macroElementsTexts = app.Settings.get("diary", "show-macro-elements-summary") ? [
+      this.getMacroElementFooterText('P', nutrition.proteins, categoryEnergyTotal),
+      this.getMacroElementFooterText('F', nutrition.fat, categoryEnergyTotal),
+      this.getMacroElementFooterText('C', nutrition.carbohydrates, categoryEnergyTotal)
+    ].filter(text => text !== '') : [];
```

Comment:
Please change the order to [fat, carbohydrates, proteins] because this is the order we also use in other places. See our global `energyMacroNutriments` list:

https://github.com/davidhealey/waistline/blob/70d5821de6c3462af1aac75f0e0f4e04a18d2472/www/index.js#L34